src/code_attribute/parser.rs

```diff
@@ -0,0 +1,241 @@
+use code_attribute::types::Instruction;
+use nom::{be_i16, be_i32, be_i8, be_u16, be_u32, be_u8, IResult, Offset};
+
+fn offset<'a>(remaining: &'a [u8], input: &[u8]) -> IResult<&'a [u8], usize> {
+    Ok((remaining, input.offset(remaining)))
+}
+
+fn align(input: &[u8], address: usize) -> IResult<&[u8], &[u8]> {
+    take!(input, (4 - address % 4) % 4)
+}
+
+pub fn code_parser(input: &[u8]) -> IResult<&[u8], Vec<(usize, Instruction)>> {
+    many0!(
+        input,
+        complete!(do_parse!(
+            address: apply!(offset, input)
+                >> instruction: apply!(instruction_parser, address)
+                >> (address, instruction)
+        ))
+    )
+}
+
+pub fn instruction_parser(input: &[u8], address: usize) -> IResult<&[u8], Instruction> {
+    switch!(input, be_u8,
+        0x32 => value!(Instruction::Aaload) |
+        0x53 => value!(Instruction::Aastore) |
+        0x01 => value!(Instruction::Aconstnull) |
+        0x19 => map!(be_u8, |index| Instruction::Aload(index)) |
```

Comment:
There are many lines here following the same pattern, mapping where the closure calls a function that takes the same parameter.

These can be simplified from:

```rust
map!(be_u8, |index| Instruction::Aload(index))
```

to

```rust
map!(be_u8, Instruction::Aload)
```

See: https://rust-lang.github.io/rust-clippy/master/#redundant_closure